fl4health/reporting/wandb_reporter.py

```diff
@@ -0,0 +1,123 @@
+from enum import Enum
+from pathlib import Path
+from typing import Any
+
+import wandb
+import wandb.wandb_run
+
+from fl4health.reporting.base_reporter import BaseReporter
+
+
+class StepType(Enum):
+    ROUND = "round"
+    EPOCH = "epoch"
+    BATCH = "batch"
+
+
+# TODO: Add ability to parse data types and save certain data types in specific ways
+# (eg. Artifacts, Tables, etc.)
+
+
+class WandBReporter(BaseReporter):
+    def __init__(self, step_type: StepType | str = StepType.ROUND, **kwargs: Any) -> None:
```

Comment:
In general I agree but in this instance I thought it'd be fine since all the kwargs go straight to the wandb.init function detailed [here](https://docs.wandb.ai/ref/python/init/). I want to expose all the arguments but didn't want to unnecessarily clutter the wandbreporter init method

Reply:
Yeah I think you're right. Perhaps we can just limit ourselves to the ones we absolutely expect to be there and the rest can be in kwargs. Perhaps we can also put a note in the documentation to where a user might find the args for the WandB object?